packages/jupyter-ai-magics/jupyter_ai_magics/embedding_providers.py

```diff
@@ -7,7 +7,7 @@
     OpenAIEmbeddings,
 )
 from langchain.embeddings.base import Embeddings
-from pydantic import BaseModel, Extra
+from langchain.pydantic_v1 import BaseModel, Extra
```

Comment:
Shouldn't our project instead declare `pydantic~=1.0` instead of importing `pydantic` from `langchain`? What is the benefit here?

I feel that we should minimize imports from `langchain` unless we are actually using features from LangChain given the project's instability.

Reply:
I agree on specifying a pydantic version in our package. Since, we are overriding LangChain providers, I was depending on LangChain's version, but ok to control it from our end.